examples/camera/third_person.rs

```diff
@@ -0,0 +1,299 @@
+//! This example shows how to implement a third person camera.
+use bevy::input::mouse::{MouseButton, MouseMotion, MouseWheel};
+use bevy::prelude::*;
+use std::f32::consts::PI;
+
+fn main() {
+    App::new()
+        .insert_resource(AmbientLight {
+            color: Color::WHITE,
+            brightness: 1.0 / 5.0f32,
+        })
+        .insert_resource(OldCursorPosition(Vec2::ZERO))
+        .add_plugins(DefaultPlugins)
+        .add_plugin(ThirdPersonCameraPlugin)
+        .add_startup_system(startup)
+        .add_system(mouse_control)
+        .add_system(
+            tp_camera_clamp
+                // In order to avoid "overshooting" when you drag the camera beyond
+                // maximum or minimum values, clamping must happen after update_tp
+                // and before update_transform.
+                //
+                // You can get the "overshooting" effect by removing after update_tp
+                // constraing and adding an after update_transform constraint, or
+                // you can get it randomly if you don't add any constraints.
+                .after(ThirdPersonCameraPlugin::update_tp)
+                .before(ThirdPersonCameraPlugin::update_transform),
+        )
+        .run();
+}
+
+/// A third person camera with a frame of reference defined by "north" and
+/// "zeroth meridian" vectors and spherical coordinates defined by the "state"
+/// vector.
+#[derive(Component, Debug)]
+struct TPCamera {
```

Comment:
_If_ we include this example I would much prefer going the explicit route and avoid shorthands

```suggestion
struct ThirdPersonCamera {
```